lib/datadog.js

```diff
@@ -82,28 +82,49 @@ var post_stats = function datadog_post_stats(payload) {
 
 var flush_stats = function datadog_post_stats(ts, metrics) {
    var counters = metrics.counters;
+   var counter_rates = metrics.counter_rates;
    var gauges = metrics.gauges;
-   var timers = metrics.timers;
-   var pctThreshold = metrics.pctThreshold;
+   var timers = metrics.timer_data;
+   var pctThresholds = metrics.pctThreshold;
+   var statsd_metrics = metrics.statsd_metrics;
 
    var host = hostname || os.hostname();
    var payload = [];
    var value;
 
    var key;
 
+   for (key in statsd_metrics) {
+     payload.push({
+        metric: get_prefix(key),
+        points: [[ts, statsd_metrics[key]]],
+        type: 'gauge',
+        host: host,
+        tags: datadogTags
+     });
+   }
+
    // Send counters
    for (key in counters) {
       value = counters[key];
-      var valuePerSecond = value / (flushInterval / 1000); // calculate 'per second' rate
+      // Fetch the pre-caculated rate
+      var valuePerSecond = counter_rates[key];
 
       payload.push({
-         metric: get_prefix(key),
+         metric: get_prefix(key) + ".per_second",
          points: [[ts, valuePerSecond]],
          type: 'gauge',
          host: host,
          tags: datadogTags
       });
+
+      payload.push({
+         metric: get_prefix(key),
+         points: [[ts, value]],
+         type: 'counter',
```

Comment:
according to this blog post, published by datadog, i think it might be better to report the type as `count`, as it would give you to option to use `as_count` and `as_rate`